cedar-policy-validator/src/level_validate.rs

```diff
@@ -0,0 +1,340 @@
+/*
+ * Copyright Cedar Contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+//! Validator for Cedar policies
+#![forbid(unsafe_code)]
+#![deny(
```

Comment:
Should't need these outside the toplevel lib.rs

Reply:
Interesting, it warned me that was the case for `#![warn(rust_2018_idioms)]` but not for these.